### ISSUE TITLE
How should responses to previous activities be modelled?

### ISSUE THREAD

**Issue by phochste** [MellonScholarlyCommunication/spec-notifications#6]
There was in COAR an interesting discussion about responses to previous activities with all kinds of subtleties involved.
E.g. respond to an as:Offer with an as:Accept or as:Reject

- How to refer to the previous as:Offer in the as:Accept | as:Reject 
- Does this reference to the previous as:Offer contain the same "validation" rules (all the MUST-s and SHOULD-s)?
- Should the previous offer be an `as:object` , `as:inReplyTo` ..something else?
- ...

**Comment by mielvds** [comment]
Depends on what you are accepting: the artefact or the offer? The latter is more intuitive, but in this context, I'd make to make every notification about an artefact. In that case, you don't need to refer to the offer?

**Comment by phochste** [comment]
I think even in case of accepting an 'artefact' you need to be able to refer to the context of the Offer.

On day 1  Alice sends to Service Hub B:

- as:Offer artefact-123 for Review
- as:Offer artefact-123 for Registration

Imagine Service Hub B can do both.

One week later Service Hub B answers:

- as:Accept artefact-123

Now Alice doesn't know what was accepted the Review or the Registration.

**Comment by mielvds** [comment]
fair point. I think the paradigm of coar-notify fits here. I'd fix the object of Accept, Reject and Undo as `as:Activity` or one of its subtypes.
